lib/shellable.ts

```diff
@@ -247,6 +249,18 @@ export class Shellable extends cdk.Construct {
     this.role = this.project.role!; // not undefined, as it's a new Project
     asset.grantRead(this.role);
 
+    // Grant read access to secrets
+    Object.entries(props.environmentSecrets ?? {}).forEach(([name, secretArn]) => {
+      const secret = Secret.fromSecretArn(this, `${name}Secret`, secretArn);
```

Comment:
question: I'm guessing we have a convention where we add the suffix of `Secret` and `Parameter` accordingly?
Is there always going to be an `ISecret` / what happens when `fromSecretArn` can't find a Secret?

do we need to validate anything before adding the grantRead?

Reply:
No convention, just needed indentifiers here. We could use the name as is I suppose.

Whats the recommended way for verifying that a `X.fromXyz()` call resolved to something?

Reply:
I'm not sure actually, but I think what you have is fine based on the examples i see in our codebase